apps/openmw/mwgui/tradewindow.cpp

```diff
@@ -361,15 +355,15 @@ namespace MWGui
             float dispositionTerm = gmst.find("fDispositionMod")->getFloat() * (clampedDisposition - 50);
             float pcTerm = (dispositionTerm - 50 + a1 + b1 + c1) * playerStats.getFatigueTerm();
             float npcTerm = (d1 + e1 + f1) * sellerStats.getFatigueTerm();
-            float x = gmst.find("fBargainOfferMulti")->getFloat() * d + gmst.find("fBargainOfferBase")->getFloat();
-            if (buying)
-                x += abs(int(pcTerm - npcTerm));
-            else
-                x += abs(int(npcTerm - pcTerm));
+            float x = gmst.find("fBargainOfferMulti")->getFloat() * percentDiff
+                    + gmst.find("fBargainOfferBase")->getFloat()
+                    + std::abs(int(npcTerm - pcTerm));
 
+            // notify and decrease merchant disposition on haggle failure
+            // note: haggle auto-fails if player tries to buy items and receive gold (see above)
             int roll = Misc::Rng::rollDice(100) + 1;
-            if(roll > x || (mCurrentMerchantOffer < 0) != (mCurrentBalance < 0)) //trade refused
-            {
+
+            if ( x < roll || (mIsPlayerSelling && !mIsMerchantBuying) ) {
```

Comment:
The original code was correct. The formula on the research page says the trade is accepted if `roll <= x`, so this line runs the barter failure code if `x < roll` or `roll > x`, which are equivalent.